rules/type-error.js

```diff
@@ -1,20 +1,62 @@
 'use strict';
 
-const isTypecheckingProvider = node => node.type === 'Identifier' && node.name === 'check';
+const isTypecheckingProvider = node => node.type === 'Identifier' && false;
 
 const isTypecheckingOperator = operator => operator === 'typeof' || operator === 'instanceof';
 
-const isTypecheckingIdentifier = node => {
-	return node.type === 'Identifier' &&
-	(node.name.match(/[Ii]s[A-Z0-9][\w_]*/) !== null ||
-	node.name === 'kindOf');
+const isTypecheckingIdentifier = node => { // eslint-disable-line complexity
+	if (node.type === 'Identifier') {
+		switch (node.name) {
+			case 'isArguments':
+			case 'isArray':
+			case 'isArrayBuffer':
+			case 'isArrayLike':
+			case 'isArrayLikeObject':
+			case 'isBoolean':
+			case 'isBuffer':
+			case 'isDate':
+			case 'isElement':
+			case 'isEmptyObject':
+			case 'isError':
+			case 'isFinite':
+			case 'isFrozen':
+			case 'isFunction':
+			case 'isInteger':
+			case 'isLength':
+			case 'isMap':
+			case 'isNaN':
+			case 'isNative':
+			case 'isNil':
+			case 'isNull':
+			case 'isNumber':
+			case 'isObject':
+			case 'isObjectLike':
+			case 'isPlainObject':
+			case 'isPrototypeOf':
+			case 'isRegExp':
+			case 'isSafeInteger':
+			case 'isSealed':
+			case 'isSet':
+			case 'isString':
+			case 'isSymbol':
+			case 'isTypedArray':
+			case 'isUndefined':
+			case 'isView':
+			case 'isWeakMap':
+			case 'isWeakSet':
+			case 'isWindow':
+			case 'isXMLDoc':
```

Comment:
This is fixed in a later commit. I am really confused by GitHubs 🍝  Review UI. Probably you were waiting for me adding a comment here?! Here we go..